### PR TITLE
add param to exclude days from week header

### PR DESCRIPTION
**Pull request by etwillbefine**
/cc @mattlewis92 

is that all? absolutely not sure about. 
relates to https://github.com/mattlewis92/angular-calendar/pull/155

-> no clue about whats going on with the tests and i'm not really in that angular/webpack topic :/ do you know whats wrong?

### CODE REVIEW

**Comment by mattlewis92** [comment]
@etwillbefine fixing it now in master for you, give me 10 mins :)

**Comment by etwillbefine** [comment]
sure, thanks

**Comment by mattlewis92** [comment]
Should be fixed now if you merge with master 😄

File: src/calendarUtils.ts [hunk 1]
```diff
@@ -188,13 +188,15 @@ const getWeekDay: Function = ({date}: {date: Date}): WeekDay => {
...
-export const getWeekViewHeader: Function = ({viewDate, weekStartsOn}: {viewDate: Date, weekStartsOn: number}): WeekDay[] => {
-
+export const getWeekViewHeader: Function = ({viewDate, weekStartsOn, excluded}:
+    {viewDate: Date, weekStartsOn: number, excluded: number[]}): WeekDay[] => {
   const start: Date = startOfWeek(viewDate, {weekStartsOn});
   const days: WeekDay[] = [];
   for (let i: number = 0; i < DAYS_IN_WEEK; i++) {
-    const date: Date = addDays(start, i);
-    days.push(getWeekDay({date}));
+    if (!excluded || !excluded.some(e => i === e)) {
```
Comment (mattlewis92) [inline src/calendarUtils.ts]: You can just use a default parameter for excluded i.e. `excluded = []` in the function definition

File: src/calendarUtils.ts [hunk 2]
```diff
@@ -272,6 +274,9 @@ export const getMonthView: Function = ({events = [], viewDate, weekStartsOn}:
   const days: MonthViewDay[] = [];
   for (let i: number = 0; i < differenceInDays(end, start) + 1; i++) {
     const date: Date = addDays(start, i);
+    if (excluded && excluded.some(e => date.getDay() === e)) {
```
Comment (mattlewis92) [inline src/calendarUtils.ts]: Same here

File: src/calendarUtils.ts [hunk 3]
```diff
@@ -272,6 +274,9 @@ export const getMonthView: Function = ({events = [], viewDate, weekStartsOn}:
...
+      continue;
```
Comment (mattlewis92) [inline src/calendarUtils.ts]: I'm not a huge fan of using `continue` in loops as it can sometimes make the logic hard to follow, can you inver the logic and add the contents below into the if statement?

File: src/calendarUtils.ts (hunk 1 above)
Comment (mattlewis92) [inline src/calendarUtils.ts]: Can you make `excluded` a default parameter instead?

File: src/calendarUtils.ts [hunk 4]
```diff
@@ -258,8 +260,8 @@ export const getWeekView: Function = ({events = [], viewDate, weekStartsOn}:
 
 };
 
-export const getMonthView: Function = ({events = [], viewDate, weekStartsOn}:
-  {events: CalendarEvent[], viewDate: Date, weekStartsOn: number})
+export const getMonthView: Function = ({events = [], viewDate, weekStartsOn, excluded}:
+  {events: CalendarEvent[], viewDate: Date, weekStartsOn: number, excluded: number[]})
   : MonthView => {
```
Comment (mattlewis92) [inline src/calendarUtils.ts]: Same here, just do `excluded = []` the same as events

**Review by mattlewis92 — CHANGES_REQUESTED** [review]

File: src/calendarUtils.ts (hunk 3 above)
Comment (etwillbefine) [reply src/calendarUtils.ts]: you are the owner, sure :)

File: src/calendarUtils.ts (hunk 1 above)
Comment (etwillbefine) [reply src/calendarUtils.ts]: tried to add it after `number[] = []` which didn't work. Works in function def. thanks

**Comment by mattlewis92** [comment]
You're also going to need to modify [the logic that sets event spans and offsets on the week view](https://github.com/mattlewis92/calendar-utils/blob/master/src/calendarUtils.ts#L111-L133) for this to work. For example:
* For a 7 day week an event starts on monday, the event offset is 1. Hiding sundays so the first day of the week is a monday, the event offset is now 0
* For an event that spans 7 days, hiding a day its span should now be 6

I'd recommend starting with some tests for these and other cases as it will get you to think about all the cases first and then modifying the code to make them pass should be relatively straight forward.

**Comment by etwillbefine** [comment]
Thanks, already thought about it and saw the current results in the demo... 
Will continue work on it 
-> Have to get a better understanding on the whole code

**Comment by mattlewis92** [comment]
Awesome, if there's anything else I can do to help let me know!

**Comment by mattlewis92** [comment]
@etwillbefine this is looking good, I don't have time to review it properly atm, in the meantime you could update your PR on the `angular-calendar` repo to use this branch by setting `"calendar-utils": "etwillbefine/calendar-utils#patch-1"` in package.json, so that you can continue your work there without being blocked by this PR being merged 😄

File: src/calendarUtils.ts [hunk 5]
```diff
@@ -80,6 +80,7 @@ export interface MonthViewDay extends WeekDay {
 export interface MonthView {
   rowOffsets: number[];
   days: MonthViewDay[];
+  daysInWeek: number;
```
Comment (mattlewis92) [inline src/calendarUtils.ts]: Can you rename this to `totalDaysVisibleInWeek` to make it a bit more verbose. At first glance I thought it was an array of visible days rather than just the count.

**Review by mattlewis92 — APPROVED** [review]

**Comment by mattlewis92** [comment]
Published as 0.40.0 🎉

**Comment by etwillbefine** [comment]
thank you very much!
